### PR TITLE
Re-enabling tests for TaskApiController

### LINKED ISSUE #1268: Re-implement / fix TaskApiController tests
1. Re-enable tests marked to be skipped during the RTM migration.
2. Update as required to ensure the tests are asserting the correct things.
3. Fix any code required to make the tests pass.

### PR DESCRIPTION
**Pull request by stevejgordon**
Fixes #1268

### CODE REVIEW

**Comment by stevejgordon** [comment]
@tonysurma If this can go in after #1270 it's probably easier. This will need a rebase but since it's three files it shouldn't be too bad.

**Comment by tonysurma** [comment]
#1270 merged so this is ready to be rebased and merged.  Thanks @stevejgordon

**Comment by tonysurma** [comment]
after rebase I will merge it @stevejgordon

**Comment by stevejgordon** [comment]
@tonysurma Struggling to get the rebase working but as soon as I do, I'll get this committed.

**Comment by stevejgordon** [comment]
@tonysurma should be good to go now.

**Comment by stevejgordon** [comment]
wow, not having a good day with this PR. Let me check again. That test was passing locally a couple of mins ago. heading out now, so will look at this later.

**Comment by stevejgordon** [comment]
@tonysurma Hopefully this time it'll all pass!

File: AllReadyApp/Web-App/AllReady.UnitTest/Controllers/TaskApiControllerTests.cs [hunk 1]
```diff
@@ -77,7 +77,7 @@ public class TaskApiControllerTests
             var mediator = new Mock<IMediator>();
             mediator.Setup(x => x.SendAsync(It.IsAny<EventByEventIdQuery>())).ReturnsAsync(new Event());
             mediator.SetupSequence(x => x.SendAsync(It.IsAny<TaskByTaskIdQuery>()))
-                .ReturnsAsync(new AllReadyTask()).ReturnsAsync(null);
+                .ReturnsAsync(allReadyTask).ReturnsAsync(null);
```
Comment (sanilpaul) [inline AllReadyApp/Web-App/AllReady.UnitTest/Controllers/TaskApiControllerTests.cs]: @stevejgordon Does this compile for you? I am getting this error at this line. Not sure whether I am missing something
    `Error	CS1061	'ISetupSequentialResult<Task<AllReadyTask>>' does not contain a definition for 'ReturnsAsync' and no extension method 'ReturnsAsync' accepting a first argument of type `

Comment (stevejgordon) [inline AllReadyApp/Web-App/AllReady.UnitTest/Controllers/TaskApiControllerTests.cs]: I had some issues initially. @mgmccarthy upgraded to a new version of MOQ to support the SetupSequence and for some reason VS was not restoring it for me. I ended up rebooting and then doing a restore before it finally got caught up and compiled.

If that fails, you might want to try a dotnet restore from the cmd line.

**Pull request merged**